server/database.sql

```diff
@@ -18,28 +18,104 @@ USE `cl_judge`;
+DROP TABLE IF EXISTS `notifications`;
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!50503 SET character_set_client = utf8mb4 */;
+CREATE TABLE `notifications` (
+  `id` varchar(45) NOT NULL,
+  `heading` varchar(110) NOT NULL,
+  `description` varchar(110) DEFAULT NULL,
+  `created_at` varchar(110) NOT NULL,
+  `link` varchar(110) DEFAULT NULL,
+  PRIMARY KEY (`id`),
+  UNIQUE KEY `id_UNIQUE` (`id`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci;
+/*!40101 SET character_set_client = @saved_cs_client */;
+
+--
+-- Table structure for table `user_groups`
+--
+
+DROP TABLE IF EXISTS `user_groups`;
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!50503 SET character_set_client = utf8mb4 */;
+CREATE TABLE `user_groups` (
+  `id` varchar(45) NOT NULL,
+  `username` varchar(45) NOT NULL,
+  `group_id` varchar(45) NOT NULL,
+  `is_group_moderator` tinyint NOT NULL,
+  PRIMARY KEY (`id`),
+  UNIQUE KEY `id_UNIQUE` (`id`),
+  KEY `user_groups_fk_username_idx` (`username`),
+  KEY `user_groups_fk_group_id_idx` (`group_id`),
+  CONSTRAINT `user_groups_fk_group_id` FOREIGN KEY (`group_id`) REFERENCES `groups` (`id`) ON DELETE CASCADE ON UPDATE CASCADE,
+  CONSTRAINT `user_groups_fk_username` FOREIGN KEY (`username`) REFERENCES `users` (`username`) ON DELETE CASCADE ON UPDATE CASCADE
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci;
+/*!40101 SET character_set_client = @saved_cs_client */;
+
+--
+-- Table structure for table `user_notifications`
+--
+
+DROP TABLE IF EXISTS `user_notifications`;
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!50503 SET character_set_client = utf8mb4 */;
+CREATE TABLE `user_notifications` (
+  `id` varchar(45) NOT NULL,
+  `username` varchar(45) NOT NULL,
+  `notification_id` varchar(45) NOT NULL,
+  PRIMARY KEY (`id`),
+  UNIQUE KEY `id_UNIQUE` (`id`),
+  KEY `user_notifications_fk_username_idx` (`username`),
+  KEY `user_notifications_fk_notification_id_idx` (`notification_id`),
+  CONSTRAINT `user_notifications_fk_notification_id` FOREIGN KEY (`notification_id`) REFERENCES `notifications` (`id`) ON DELETE CASCADE ON UPDATE CASCADE,
+  CONSTRAINT `user_notifications_fk_username` FOREIGN KEY (`username`) REFERENCES `users` (`username`) ON DELETE CASCADE ON UPDATE CASCADE
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci;
+/*!40101 SET character_set_client = @saved_cs_client */;
+
+--
+-- Table structure for table `users`
+--
+
+DROP TABLE IF EXISTS `users`;
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!50503 SET character_set_client = utf8mb4 */;
+CREATE TABLE `users` (
   `username` varchar(45) NOT NULL,
   `secret` longtext NOT NULL,
   `full_name` varchar(45) NOT NULL,
   `admission_number` varchar(45) NOT NULL,
-  `is_admin` tinyint DEFAULT NULL,
   `email` varchar(45) NOT NULL,
   `mobile` varchar(45) NOT NULL,
-  `logged_in` tinyint DEFAULT NULL,
+  `is_admin` tinyint DEFAULT NULL,
```

Comment:
`is_admin` in users table for super-users like profs
By default null...

A separate API is to be built for making `is_admin = true`....the person signing up is required to upload some id proof or something (of being a prof), which will be verified by maintainers and then the person will be can be enabled as admin

Reply:
We can do it mannually.